lib/msf/ui/debug.rb

```diff
@@ -0,0 +1,253 @@
+
+        str = "Framework Major Version: #{major} \n"
+        str += "Framework Minor Version: #{minor} \n"
+        str += "Framework Patch Version: #{patch} \n"
+        str += "Ruby Version: #{RUBY_DESCRIPTION} \n"
+        str += "DB Session Status: #{db_connection_info(framework)} \n"
+        str += "Installation Method: #{installation_method} \n"
+        str += "\n"
+        str += "OS: #{Gem::Platform.local.os} \n"
+        str += "OS Version: #{Gem::Platform.local.version} \n"
+        str += "CPU Architecture: #{Gem::Platform.local.cpu} \n"
+
+        build_section('Version/Install', 'The versions & install method of your Metasploit setup:', str)
+      end
+
+      class << self
+
+        private
+
+        def add_hash_to_ini_group(ini, hash, group_name)
+          if hash.empty?
+            return
+          end
+
+          unless ini.group?(group_name)
+            ini.add_group(group_name)
+          end
+
+          hash.each_pair do |k, v|
+            ini[group_name][k] = v
+          end
+        end
+
+        def concat_str_array_from_last_idx(array, concat_total, extra_padding = false)
+          end_pos = array.length > concat_total ? array.length - concat_total : 0
+          start_pos = array.length - 1
+
+          str = ''
+          while start_pos >= end_pos
+            str += array[start_pos]
+            str += "\n\n" if extra_padding
+            start_pos -= 1
+          end
+
+          str.strip
+        end
+
+        # Copy pasta of the print_connection_info method in console/command_dispatcher/db.rb
+        def db_connection_info(framework)
+          unless framework.db.connection_established?
+            return "#{framework.db.driver} selected, no connection"
+          end
+
+          cdb = ''
+          if framework.db.driver == 'http'
+            cdb = framework.db.name
+          else
+            ::ActiveRecord::Base.connection_pool.with_connection do |conn|
+              if conn.respond_to?(:current_database)
+                cdb = conn.current_database
+              end
+            end
+          end
+          output = "Connected to #{cdb}. Connection type: #{framework.db.driver}."
+          output += " Connection name: #{framework.db.get_data_service}." if framework.db.get_data_service
+
+          output
+        end
+
+        def build_section(header_name, blurb, content)
+          <<~SECTION
+            ##  %grn#{header_name.strip}%clr
+            #{blurb.strip}
+            #{with_collapsible_wrapper(content.strip)}
+
+          SECTION
+        end
+
+        def with_collapsible_wrapper(content)
+          <<~WRAPPER
+            <details>
+            <summary>Collapse</summary>
+
+            ```
+            #{content}
+            ```
+
+            </details>
+          WRAPPER
+        end
+
+        def installation_method
+          if File.exist?(File.join(File::SEPARATOR, 'opt', 'metasploit-framework', 'version.yaml'))
```

Comment:
`Msf::Config.install_root` gives you the install folder, and it seems to be a `.yml` file extension:

```
File.exist?(File.join(Msf::Config.install_root, 'version.yml'))
```

Reply:
The existing approach wouldn't work for windows either :eyes: